server/api/lib/trees/hod_hasharon_tree_permit.js

```diff
@@ -0,0 +1,141 @@
+const proxy = require('./../proxy');
+const cheerio = require('cheerio');
+const Config = require('../../lib/config');
+const TreePermit = require('../../model/tree_permit');
+const {
+  REGIONAL_OFFICE, PERMIT_NUMBER, APPROVER_TITLE, ACTION,
+  LAST_DATE_TO_OBJECTION, TOTAL_TREES, REASON_SHORT,
+  PLACE, STREET,
+  TREES_PER_PERMIT, PERSON_REQUEST_NAME, TREE_PERMIT_URL,
+} = require('../../model/tree_permit_constants');
+const { formatDate } = require('./utils');
+const Log = require('../log');
+
+const TREES_HOD_HASHARON_URL = Config.get('trees.hodHasharonUrl');
+const hodHashTreePermit = {
+  urls: [TREES_HOD_HASHARON_URL]
+};
+
+async function parseTreesHtml(url) {
+  const treesHtml = await proxy.get(url);
+  const dom = cheerio.load(treesHtml, {
+    decodeEntities: false
+  });
+  if (!dom) {
+    console.error('cheerio dom is null');
+  }
+  const keys = [];
+  const result = [];
+  dom('.page-content-body').find('table').find('tr').each((row, elem) => {
+    if (row === 0) {
+      dom(elem).find('th').each((idx, elem) => {
+        const key = dom(elem).text().trim();
+        keys.push(key);
+      });
+      return;
+    }
+    const treePermit = {};
+    dom(elem).find('td,th').each((idx, elem) => {
+            const url = idx === 6 ? dom(elem).find('a').attr('href') : '';
+            if (url.length > 0) {
+                treePermit['url'] = url.substr(1);
+            }
+      const value = dom(elem).text().trim();
+      const key = keys[idx];
+      treePermit[key] = value;
+    });
+    result.push(treePermit);
+  });
+  Log.info(`number of hod hasharon permits: ${result.length}`);
+  return result;
+}
+
+function processRawPermits(rawPermits) {
+  try {
+    const treePermits = rawPermits.map(raw => {
+      try{
+        
+        const street = raw['שם הרחוב'];
+        
+        const last_date_to_objection = formatDate(raw['ניתן להגיש ערעור עד'], '09:00', 'DD.MM.YY');
+        if (!last_date_to_objection) {
+          Log.error(`No / Bad dates format, ignore this license: hod hasharon, ${raw['שם הרחוב']} , ${raw['ניתן להגיש ערעור עד']}`);
+          return null;
+        }
+        
+        const treesPerPermit = parseTreesPerPermit(raw['סוג העצים'], raw['מספר עצים']);
```

Comment:
Use of Hebrew strings inside code is very problematic. It hurts the maintainability of the code. It hard to edit in some editors.
Can you please maintain a list of constants and use them everywhere you currently have Hebrew string? If I am not mistaken, this is how it is done elsewhere.

Reply:
moved